scripts/complete_liquidity_provision.js

```diff
@@ -105,15 +116,29 @@ module.exports = async (callback) => {
       callback("Error: Choose a smaller fleetSize, otherwise your payload will be to big for Infura nodes")
     }
 
-    console.log(`2. Deploying ${argv.fleetSize} trading brackets`)
-    const bracketAddresses = await deployFleetOfSafes(masterSafe.address, argv.fleetSize, true)
-    console.log("List of bracket traders in one line:", bracketAddresses.join())
+    let bracketAddresses
+    if (argv.brackets.length > 0) {
+      console.log("==> Skipping safe deployment and using brackets safeOwners")
+      bracketAddresses = argv.brackets
+      // Ensure that safes are all owned solely by masterSafe
+      for (const safeAddr of argv.brackets) {
+        const safeOwners = await (await GnosisSafe.at(safeAddr)).getOwners()
+        // TODO: write function like safeOwnedSolelyBy(safe, owner) -> Bool
```

Comment:
Like the function `isOnlySafeOwner` from `trading_strategy_helpers.js`? :stuck_out_tongue:

Reply:
I have another problem with this. Because I was hoping to avoid "fetching" the master safe again. Unfortunately, this `isOnlySafeOwner` only takes addresses and has to fetch both safes... Is there something that can be done about this or it is really worth the extra fetch?